thamos/lib.py

```diff
@@ -97,7 +101,30 @@ def wrapper(*args, **kwargs):
     return wrapper
 
 
-def _wait_for_analysis(status_func: Callable[..., Any], analysis_id: str) -> None:
+def get_analysis_status(analysis_id: str, source: str) -> bool:
+    """Handle the the multiple response types available while asking for result of a anaylsis result."""
+    config = Configuration()
+    host = thoth_config.explicit_host
+    if not host:
+        thoth_config.load_config()
+        host = thoth_config.content.get("host") or config.host
+    source_url = _SOURCE.get(source)
+    response = requests.Session().get(
+        f"https://{host}/api/v1/{source_url}/{analysis_id}"
+    )
+    if response.status_code == 202:
+        return False
+    # Return true if result is ready.
+    elif response.status_code == 200:
+        return True
+    else:
+        response.raise_for_status()
+    return False
```

Comment:
Raising some exception here would do the trick. Or at least logging a warning so users know there is something wrong.

Reply:
`.raise_for_status()` raises an exception right?

Reply:
Only if an error indicating HTTP status code is returned.

Reply:
Thanks I have raised an exception